httpcore5/src/main/java/org/apache/hc/core5/util/TimeValue.java

```diff
@@ -274,8 +276,8 @@ public TimeUnit getTimeUnit() {
     @Override
     public int hashCode() {
         int hash = LangUtils.HASH_SEED;
-        hash = LangUtils.hashCode(hash, duration);
-        hash = LangUtils.hashCode(hash, timeUnit);
+        hash = LangUtils.hashCode(hash, this.convert(TimeUnit.NANOSECONDS));
+        hash = LangUtils.hashCode(hash, TimeUnit.NANOSECONDS);
```

Comment:
@michael-o `hash = LangUtils.hashCode(hash, TimeUnit.NANOSECONDS)` can be dropped as unnecessary.